util/stopper.go

```diff
@@ -51,17 +50,20 @@ type Stopper struct {
 	stopped  chan struct{}  // Closed when stopped completely
 	stop     sync.WaitGroup // Incremented for outstanding workers
 	mu       sync.Mutex     // Protects the fields below
```

Comment:
final nit: I believe you can remove this and embed `drain` by value, and then lock via `drain.L.Lock()`